json_rpc/router.nim

```diff
@@ -59,48 +61,57 @@ proc wrapError*(code: int, msg: string, id: JsonNode = newJNull(),
       $id, $code, escapeJson(msg), $data
     ] & "\r\n")
 
-proc route*(router: RpcRouter, node: JsonNode): Future[StringOfJson] {.async, gcsafe.} =
+proc hasReturnType(params: NimNode): bool =
+  if params != nil and params.len > 0 and params[0] != nil and
+    params[0].kind != nnkEmpty:
+    result = true
+
+proc route*(router: RpcRouter, node: JsonNode): Future[RpcResult] {.async, gcsafe.} =
```

Comment:
The change to `RpcResult` is also an API-breaking change that will upset existing consumers

Reply:
Is the `router` used outside of `nim-json-rpc`?

Reply:
Well, it's public, so it can be used - this is a bit of a messy aspect of Nim, the module system is shifty - a first litmus test would be to check if nimbus-eth1 and nimbus-eth2 still compile after the change - if these to projects compile, it's *mostly* fine for a library like json-rpc where we don't yet carefully maintain backwards compat